docker/src/main/java/brooklyn/entity/container/docker/registry/DockerRegistryImpl.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.ssh.BashCommands;
+import org.apache.brooklyn.util.time.Duration;
+
+import brooklyn.entity.container.DockerAttributes;
+import brooklyn.entity.container.docker.DockerHost;
+import brooklyn.entity.container.docker.DockerInfrastructure;
+import brooklyn.entity.container.docker.application.VanillaDockerApplicationImpl;
+
+public class DockerRegistryImpl extends VanillaDockerApplicationImpl implements DockerRegistry {
+
+    private volatile HttpFeed httpFeed;
+
+    private String getSSHMachineInstallDir() {
+        return config().get(DockerRegistry.DOCKER_HOST).sensors().get(SoftwareProcess.INSTALL_DIR);
+    }
+
+    @Override
+    public void connectSensors() {
+        super.connectSensors();
+        connectServiceUpIsRunning();
+
+        HostAndPort hostAndPort = BrooklynAccessUtils.getBrooklynAccessibleAddress(this, this.getConfig(DockerRegistry.DOCKER_REGISTRY_PORT));
+        sensors().set(Attributes.MAIN_URI, URI.create("https://" + hostAndPort + "/v2"));
+
+
+        httpFeed = HttpFeed.builder()
+                .entity(this)
+                .period(Duration.seconds(3))
+                .baseUri(getAttribute(Attributes.MAIN_URI))
+                .poll(new HttpPollConfig<Boolean>(Attributes.SERVICE_UP)
+                        .onSuccess(Functions.constant(true))
+                        .onFailureOrException(Functions.constant(false)))
+                .poll(new HttpPollConfig<String[]>(DockerRegistry.DOCKER_REGISTRY_CATALOG)
+                        .suburl("/_catalog")
+                        .onSuccess(HttpValueFunctions.jsonContents("repositories", String[].class)))
+                .build();
+    }
+
+    @Override
+    public void disconnectSensors() {
+        super.disconnectSensors();
+        disconnectServiceUpIsRunning();
+        if (httpFeed != null) {
+            httpFeed.stop();
+        }
+    }
+
+
+    @Override
+    public void init() {
+        super.init();
+
+        DockerHost host = (DockerHost) config().get(DOCKER_HOST);
+
+        config().set(DockerAttributes.DOCKER_PORT_BINDINGS, MutableMap.of(config().get(DOCKER_REGISTRY_PORT), 5000));
+
+        String installDir = host.sensors().get(SoftwareProcess.INSTALL_DIR);
+        config().set(DockerAttributes.DOCKER_HOST_VOLUME_MAPPING, MutableMap.of(Os.mergePaths(installDir, "certs"), "/certs"));
+
+        // set location customizer to new inner class
+        // customizer does the SshMachineLocation stuff below
+        // is passed a DockerHostLocation, can get to SshMachineLocation from there
+
+
+        SshMachineLocation sshMachine = host.getDynamicLocation().getMachine();
+        sshMachine.installTo("classpath://brooklyn/entity/container/docker/registry/docker-registry-create-certs.sh", Os.mergePaths(getSSHMachineInstallDir(), "docker-registry-create-certs.sh"));
```

Comment:
Shove `getSSHMachineInstallDir()` in a local variable (`dockerInstallDir`) and make `docker-registry-create-certs.sh` a constant as well